File: Frontend/src/models/Task.js
export default class Task {
  constructor(data) {
    this.id = data.id;
    this.title = data.title;
    this.description = data.description;
    this.status = data.status;
    this.assignedToUserId = data.assignedToUserId;
    this.createdAt = data.createdAt;
    this.updatedAt = data.updatedAt;
  }

  isCompleted() {
    return this.status === 'Done';
  }

  getNextStatus() {
    if (this.status === 'ToDo') return 'InProgress';
    if (this.status === 'InProgress') return 'Done';
    return null;
  }
}

File: Backend/ApiBackend/Models/Task.cs
namespace ApiBackend.Models
{
    public enum TaskStatus { ToDo, InProgress, Done }
    public class Task
    {
        public int Id { get; set; }
        public string Title { get; set; } = null!;
        public string? Description { get; set; }
        public TaskStatus Status { get; set; } = TaskStatus.ToDo;
        public int? AssignedToUserId { get; set; }
        public User? AssignedToUser { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        public List<TaskHistory> TaskHistories { get; set; } = new();
    }
}
